app/models/user.rb

```diff
@@ -25,29 +62,63 @@ class User < ApplicationRecord
   # Validate the attached image is image/jpg, image/png, etc
   validates_attachment_content_type :avatar, content_type: %r{\Aimage\/.*\Z}
 
+  validates_presence_of :phone
+
   def name
     firstname + " " + lastname
   end
 
-  def hon3y
-    median(ratings)
+  # Methods to make Devise happy
+
+  # def provider
+  #   'phone'
+  # end
+  #
+  # def uid
+  #   phone + created_at.to_s
+  # end
+  #
+  # def encrypted_password
+  #   phone
+  # end
+  #
+  # def encrypted_password_changed?
+  #   false
+  # end
+
+  # def confirmation_token
+  #   pin
+  # end
+  #
+  # def confirmation_token=(token)
+  #   self.pin = token
+  # end
+  #
+  # def confirmation_sent_at=(timestamp)
+  # end
+
+  # end Devise methods
+
+  def generate_pin!
+    self.pin = rand(0000..9999).to_s.rjust(4, "0")
```

Comment:
Can we use [`SecureRandom.random_number(9999)`](https://ruby-doc.org/stdlib-2.6.5/libdoc/securerandom/rdoc/SecureRandom.html) instead?